### PR TITLE
Use testify for AWS provider

### PR DESCRIPTION
**Pull request by chbatey**
We've started using testify for some private work we plan to raise PRs for. This PR includes an example test to see if others are happy for testify asserts and mocks to be used in aws provider tests. At present we're finding it hard to test the exact calls we're making to AWS.

This has two main advantages:
- The use of the mock package to verify API calls against the aws SDK
- Nicer error messages for asserts without having to use if statements

Example error message for an assertion failure:

```
Error Trace:    aws_test.go:958
Error:      Expected nil, but got: &errors.errorString{s:"Invalid format for AWS volume (aws://us-east-1a/VolumeName)"}
Messages:   Error creating Volume &{%!w(string=Invalid format for AWS volume (aws://us-east-1a/VolumeName))}
```

And for a failed verification against the AWS SDK:

```
mock: Unexpected Method Call
-----------------------------

DescribeVolumes(*ec2.DescribeVolumesInput)
        0: {
  VolumeIds: ["WrongVolumeId"]
}

The closest call I have is: 

DescribeVolumes(*ec2.DescribeVolumesInput)
        0: {
  VolumeIds: ["vol-VolumeId"]
}
```

### CODE REVIEW

**Comment by chbatey** [comment]
I signed it

**Comment by davidopp** [comment]
@justinsb

File: pkg/cloudprovider/providers/aws/aws_test.go [hunk 1]
```diff
@@ -1161,3 +1167,25 @@ func TestFindInstancesByNodeName(t *testing.T) {
...
+	awsServices.ec2.On("DescribeVolumes", expectedVolumeRequest).Return([]*ec2.Volume{
+		{
+			VolumeId:         volumeId,
+			AvailabilityZone: &awsServices.availabilityZone,
+		},
+	})
+
+	labels, err := c.GetVolumeLabels(*volumeId)
+
+	assert.Nil(t, err, "Error creating Volume %w", err)
```
Comment (justinsb) [inline pkg/cloudprovider/providers/aws/aws_test.go]: Is `%w` right here?

**Comment by justinsb** [comment]
I like it - it seems like an improvement over raw test framework, and I think we could greatly improve our AWS test coverage practically.

Not sure if people more knowledgable about go testing have any stronger opinions on testify, but I'm happy to try it out in aws-land.

So no objection here (and LGTM barring the nit).

File: pkg/cloudprovider/providers/aws/aws_test.go (hunk 1 above)
Comment (chbatey) [inline pkg/cloudprovider/providers/aws/aws_test.go]: Should be %v, will fix.

**Comment by chbatey** [comment]
Updated for the nit @justinsb

**Comment by davidopp** [comment]
LGTM

(I did not review it, but Justin said LGTM except for nit, which I verified was addressed.)

**Pull request merged**